### DISCUSSION TITLE
how to set NEXTAUTH_URL environment variable in Cloudflare deploy.

### DISCUSSION THREAD

**Discussion by oyeon-kwon** [nextauthjs/next-auth#8195] [Help]
### Question 💬

I deployed the app to cloudflare using next auth. Like vercel, cloudflare provides a preview deployment, where the url is set randomly. (ex. https://2cc1cc59.na-lab.pages.dev).

So even if I set the NEXTAUTH_URL environment variable, I get 404 [CLIENT_FETCH_ERROR]
 return an error.

I recognized through the official documentation (https://next-auth.js.org/deployment)  that NEXTAUTH_URL does not have to be entered when distributing to vercel and netlify.

However, I am curious about how to enter the NEXTAUTH_URL environment variable in **cloudflare**.

### How to reproduce ☕️

https://2cc1cc59.na-lab.pages.dev/

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by stale** [comment]
It looks like this issue did not receive any activity for 60 days. It will be closed in 7 days if no further activity occurs. If you think your issue is still relevant, commenting will keep it open. Thanks!

**Comment by tdack** [comment]
You can deploy to a preview environment with Cloudflare Pages and then your url would be `preview.na-lab.pages.dev`.

You can then deploy with:
`wrangler pages deploy <path to build> --commit-dirty=true --branch=preview --project-name=na-lab`
